ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitter.scala

```diff
@@ -455,6 +473,36 @@ private[kvutils] class TransactionCommitter(
           RejectionReason.Inconsistent("Causal monotonicity violated")))
   }
 
+  /**
+    * Validates that 'contract key' -> 'contract id' mapping resolved by a participant is still valid.
+    */
+  private def validateContractKeysConsistency(
+      recordTime: Option[Timestamp],
+      transactionEntry: DamlTransactionEntrySummary,
+      resolvedKeys: Map[DamlContractKey, String]
+  )(): StepResult[DamlTransactionEntrySummary] = {
+    val keysAreConsistent = transactionEntry.contractKeyToIdMap.forall {
+      case (contractKey, submittedContractId) =>
+        resolvedKeys.get(contractKey) match {
+          case None =>
+            // This means that there is no such key yet in the context.
+            // We allow such submission because the submitted contractKey is ~most likely~ a result of a create command,
+            // however a more strict validation should be done here to verify that the command type was really create.
+            true
+          case contractId: Some[String] =>
```

Comment:
That's not how it should be. If the transaction first creates a contract with a key, then we want to return `Inconsistent` if there is already a contract with the same key. In that case, we must return `false`. So the default case should also be fine for the `None` case here.

```suggestion
            // We report any contract key resolution mismatch as `RejectionReason.Inconsistent`
            // even if the submitted resolution is obviously wrong. A more strict validation should
            // be done here to verify that the submitted contract ID refers to a contract with the `contractKey`
            // as its key.
```

Reply:
Fixed.